modules/dag-l1/src/main/scala/org/tessellation/dag/l1/StateChannel.scala

```diff
@@ -77,12 +79,14 @@ class StateChannel[F[_]: Async: KryoSerializer: SecurityProvider: Random](
         appConfig.consensus.tipsCount
       )
     }
+    .evalTap(_ => semaphores.blockCreation.release)
```

Comment:
I'd wrap `canStartOwnConsensus` and what's inside `evalMap` in a resource-fashion (`MonadCancel.uncancelable` or `.guarantee`), to be 100% sure the semaphore will be released. I have a gut feeling that this `evalTap` may not be executed if evalMap will throw an error, causing infinite semaphore acquisition.

Reply:
Makes sense but also if anything throws in here the stream will close, so we are doomed either way I guess. In general we need to always handle errors if we don't want the stream to close.

Reply:
And handling errors in evalMap also gives us security regarding the semaphore acquisition.

Reply:
We'll, I think no. Stream errors can be handled in many ways. One if it is a retry. And we don't know how errors are handled here in that place, that's why it's crucial to make safe semaphore operations and be 100% sure that the semaphore will be released when acquired, no matter what.